src/sources.js

```diff
@@ -356,11 +357,16 @@ class S3Source extends Source {
     return `https://${this.bucket}.s3.amazonaws.com`;
   }
   async _listObjects() {
-    return new Promise((resolve, reject) => {
+    if (!AWS_CREDENTIALS_PROVIDED) {
+      utils.warn(`Cannot list S3 objects for group '${this.name}' as no AWS credentials supplied.`);
+      return [];
+    }
+    return new Promise((resolve) => {
       let contents = [];
       S3.listObjectsV2({Bucket: this.bucket}).eachPage((err, data, done) => {
         if (err) {
-          return reject(err);
+          utils.warn(`Could not list S3 objects for group '${this.name}'\n${err.message}`);
+          return resolve([]);
```

Comment:
I am not very familiar with the codebase, but am trying to be useful :)

Where will these warnings be printed and who will be reading them in the end?

Shouldn't this and the above check `reject()` the promise, and the error will be handled in the caller function instead? Or, even better, allow the exceptions to propagate all the way to the request handler of the server or the main function of the CLI and handle all the errors there? (depending on how much we want to unwind the stack)

Reply:
Always useful comments @ivan-aksamentov ! 

These changes are primarily for development purposes, where one may forget to supply (valid) AWS credentials. The messages will be printed in terminal running the server. We could indeed `reject` here which would, after the recent updates to our handers, result in a 404 or similar response. Would you think this is preferable to the current implementation, which returns 200 but with no datasets/narratives?

P.S. Our "live" instances of the server (nextstrain.org, dev.nextstrain.org, Heroku Review Apps) all have AWS credentials set as config variables.

Reply:
Oh, I see. I misunderstood the feature. I thought it's user-facing. If for dev only, then it's fine, I guess.

The advantage of replying with 4xx (and I would carry over the whatever code AWS replies with, probably 403) is that the client can then show a specific message, as opposed to an empty page. Depending on want we want here. My understanding is that we want an empty result in dev, right?

Reply:
> My understanding is that we want an empty result in dev, right?

Initially, I was going to make the server die with an error if credentials weren't supplied (at start time), as for nextstrain devs it indicates "you forgot to set the appropriate env!", but that seemed overly nasty for people who have forked the repo etc.

I will rebase this PR upon our new server handling (once the fixes in #327 have gone in) and test what happens if we reject here, as that seems like the most consistent approach.